app/controllers/events_controller.rb

```diff
@@ -9,10 +9,13 @@ def new
 
   def create
     @event = Event.new(event_params)
-    @event.trail_id = trail_params_id
-    @event.user_id = current_user.id
-    @event.save
-    redirect_to event_path(@event)
+    if @event.save
+      current_user.hosting << @event
```

Comment:
Hosting comes from the aliasing that happens on the User model now. You can see it defined in the has many through relationship.